PF/PF-Assignment2/FixedProductPromotion.java

```diff
@@ -0,0 +1,151 @@
+	}
+	
+	/*
+	 * @param   item id (long), promo code (string)
+	 * @return  minimum item price
+	 */
+	public double getMinimumPrice(long itemID, String code) {
+		
+		this.setMinimumPrice(itemID, code);
+		
+		return this.getPrice(itemID);		
+	}
+	
+	/*
+	 * @param  item id (long), promo code (String)
+	 * @return nothing
+	 * set minimum price for item after apply promo code discount and fixed discount
+	 */
+	public void setMinimumPrice(long itemID, String code) {
+		double priceAfterFixedDiscount ;
+		double price = this.getPrice(itemID);
+		
+		priceAfterFixedDiscount = price - (price * (this.fixedDiscount.get(itemID)/100.0));
+		
+		if (this.isPromotionApplicable(itemID, code)) {
+			
+			double priceAfterPromoDiscount ;
+			priceAfterPromoDiscount = price - (price *(this.getPromoDiscount(code)/100.0));
+			
+			if (priceAfterFixedDiscount < priceAfterPromoDiscount) {
+				this.setPrice(itemID, priceAfterFixedDiscount);
+			}
+			else {
+				this.setPrice(itemID, priceAfterPromoDiscount);
+			}
+		}
+	}
+	
+	/*
+	 * @param   Promotion code 
+	 * @return  discount on given promotion code
+	 */
+	public double getPromoDiscount(String code) {
+		
+		final double noDiscount = 0.0;
+		HashMap<String, Double> discount = new HashMap<>();
+		discount.put("sky10", 15.00);
+		discount.put("algo10", 10.00);
+		discount.put("laptop10", 25.00);
+		discount.put("samsung10", 40.00);
+	
+		if(discount.containsKey(code)) {
+			return discount.get(code);
+		}
+		else {
+			return noDiscount;
+		}
+		
+	}
+	
+	/*
+	 * @param   item id (int)
+	 * @return  fixed Discount on item
+	 */
+	public double getFixedDiscount(long itemID) {
+		
+		double noDiscount = 0.0;
+		
+		if(this.fixedDiscount.containsKey(itemID)) {
+			return this.fixedDiscount.get(itemID);
+		}
+		else {
+			return noDiscount;
+		}
+		
+	}
+	
+	/*
+	 * @param   item id (int), discount (double)
+	 * @return  nothing
+	 * Set fixed discount on item 
+	 */
+	public void setFixedDiscount(long itemID, double discount) {
+		
+		this.fixedDiscount.put(itemID, discount);
+		
+	}
+	
+	/*
+	 * @param   item id (int), promo code (String)
+	 * @return  true if promo code applicable otherwise false
+	 */
+	public boolean isPromotionApplicable (long itemID, String code) {
+		PromotionEnum [] promoEnum = PromotionEnum.values();
+		for(PromotionEnum promo : promoEnum) {
+			if(code.equals(promo.getCode())) {
+				Date todayDate = new Date();
+				try {
+					Date startDate = new SimpleDateFormat("dd/MM/yyyy").parse(promo.getStartDate());
+					Date endDate = new SimpleDateFormat("dd/MM/yyyy").parse(promo.getEndDate());
+					if ((startDate.compareTo(todayDate) <= 0) && (endDate.compareTo(todayDate) >= 0)) {
+						return true;
+					}
+				}
+				catch(ParseException e) {
+					System.out.println(e);
+				}
+				
+			}
+		}
+		return false;
+	}
+	
+	public static void main(String args[]) {
+		FixedProductPromotion prodPromo = new FixedProductPromotion();
+		prodPromo.add(1, "SkyBag", 1156.36, 1);
```

Comment:
Are you sure ? why add ?

Reply:
ShoppingCart will have a variable of type Promotion. Add method is to add items in cart.